servicebus/data-plane/azure-servicebus/src/main/java/com/microsoft/azure/servicebus/primitives/RequestResponseLink.java

```diff
@@ -1008,9 +848,8 @@ private void runSendLoop()
                     }
 
                     RequestResponseWorkItem requestToBeSent = this.parent.pendingRequests.get(requestIdToBeSent);
-                    if(requestToBeSent != null)
-                    {
-                        Delivery delivery = this.sendLink.delivery(UUID.randomUUID().toString().getBytes());
+                    if (requestToBeSent != null) {
+                        Delivery delivery = this.sendLink.delivery(UUID.randomUUID().toString().getBytes(UTF_8));
```

Comment:
[SpotBugs-P1] Reliance on default encoding: Found a call to a method which will perform a byte to String (or String to byte) conversion, and will assume that the default platform encoding is suitable.